### ISSUE TITLE
Uncaught error Cannot read property 'blue' of undefined

### ISSUE THREAD

**Issue by levi-l-damian** [sanity-io/sanity#2507]
Everything worked fine for couple months and suddenly it is no longer working

**Describe the bug**

Uncaught error
Cannot read property 'blue' of undefined

Stack:

TypeError: 
    at https://369leh.vercel.app/static/js/app.bundle.js?de392c25b71c5062dc63:78:25696
    at Array.reduce (<anonymous>)
    at Object.<anonymous> (https://369leh.vercel.app/static/js/app.bundle.js?de392c25b71c5062dc63:78:25658)
    at l (https://369leh.vercel.app/static/js/vendor.bundle.js?4f7e199f796bdcb9a3f5:1:410)
    at Object.<anonymous> (https://369leh.vercel.app/static/js/app.bundle.js?de392c25b71c5062dc63:78:24173)
    at l (https://369leh.vercel.app/static/js/vendor.bundle.js?4f7e199f796bdcb9a3f5:1:410)
    at Object.<anonymous> (https://369leh.vercel.app/static/js/app.bundle.js?de392c25b71c5062dc63:78:23935)
    at l (https://369leh.vercel.app/static/js/vendor.bundle.js?4f7e199f796bdcb9a3f5:1:410)
    at Object.<anonymous> (https://369leh.vercel.app/static/js/app.bundle.js?de392c25b71c5062dc63:78:23739)
    at l (https://369leh.vercel.app/static/js/vendor.bundle.js?4f7e199f796bdcb9a3f5:1:410)

(Your browsers Developer Tools may contain more info)
Uncaught TypeError: Cannot read property 'blue' of undefined
    at app.bundle.js?de392c25b71c5062dc63:78
    at Array.reduce (<anonymous>)
    at Object.<anonymous> (app.bundle.js?de392c25b71c5062dc63:78)
    at l (vendor.bundle.js?4f7e199f796bdcb9a3f5:1)
    at Object.<anonymous> (app.bundle.js?de392c25b71c5062dc63:78)
    at l (vendor.bundle.js?4f7e199f796bdcb9a3f5:1)
    at Object.<anonymous> (app.bundle.js?de392c25b71c5062dc63:78)
    at l (vendor.bundle.js?4f7e199f796bdcb9a3f5:1)
    at Object.<anonymous> (app.bundle.js?de392c25b71c5062dc63:78)
    at l (vendor.bundle.js?4f7e199f796bdcb9a3f5:1)

**To Reproduce**

1. Go to https://369leh.vercel.app/

**Expected behavior**

I was expecting to be asked for login.

Run `sanity versions` in the terminal and copy-paste the result here.
@sanity/cli                 2.10.0 (up to date)
@sanity/base                2.10.2 (up to date)
@sanity/cli                 2.10.0 (up to date)
@sanity/code-input           2.9.0 (up to date)
@sanity/components           2.2.6 (up to date)
@sanity/core                2.10.2 (up to date)
@sanity/default-layout      2.10.2 (up to date)
@sanity/default-login        2.8.0 (up to date)
@sanity/desk-tool           2.10.4 (up to date)
@sanity/production-preview   2.2.6 (up to date)
@sanity/vision              2.10.0 (up to date)
**What operating system are you using?**

**Which versions of Node.js / npm are you running?**

Run `npm -v && node -v` in the terminal and copy-paste the result here.
6.14.13
v14.17.0

**Additional context**

It may be related, or not, not sure, I can access the studio in development using localhost:3333, but the preview feature is not working.
When clicking on "Open Preview" I am getting the error: Internal Server Error
And here are the details in the console:

Loaded env from /Users/levi/react_projects/test/369leh/frontend/.env.local
ready - started server on 0.0.0.0:3000, url: http://localhost:3000
event - compiled successfully
event - build page: /
wait  - compiling...
event - compiled successfully
Using the Sanity client without specifying an API version is deprecated. See https://docs.sanity.io/help/js-client-api-version
event - build page: /api/blogs
wait  - compiling...
event - compiled successfully
event - build page: /api/preview
wait  - compiling...
event - compiled successfully
ClientError: Unauthorized - Session not found
    at onResponse (/Users/levi/react_projects/test/369leh/frontend/node_modules/@sanity/client/lib/http/request.js:27:13)
    at applyMiddleware (/Users/levi/react_projects/test/369leh/frontend/node_modules/get-it/lib-node/util/middlewareReducer.js:14:23)
    at onResponse (/Users/levi/react_projects/test/369leh/frontend/node_modules/get-it/lib-node/index.js:81:22)
    at /Users/levi/react_projects/test/369leh/frontend/node_modules/get-it/lib-node/index.js:48:55
    at callback (/Users/levi/react_projects/test/369leh/frontend/node_modules/get-it/lib-node/request/node-request.js:57:46)
    at /Users/levi/react_projects/test/369leh/frontend/node_modules/get-it/lib-node/request/node-request.js:141:14
    at DestroyableTransform.<anonymous> (/Users/levi/react_projects/test/369leh/frontend/node_modules/simple-concat/index.js:8:13)
    at Object.onceWrapper (events.js:482:28)
    at DestroyableTransform.emit (events.js:388:22)
    at endReadableNT (/Users/levi/react_projects/test/369leh/frontend/node_modules/through2/node_modules/readable-stream/lib/_stream_readable.js:1010:12)
    at processTicksAndRejections (internal/process/task_queues.js:82:21) {
  response: {
    body: {
      errorCode: 'SIO-401-ANF',
      message: 'Session not found',
      statusCode: 401,
      error: 'Unauthorized'
    },
    url: "https://ulym7tod.api.sanity.io/v1/data/query/production?query=*%5B_type%20%3D%3D%20%22blog%22%20%26%26%20slug.current%20%3D%3D%20%24slug%5D%20%7B%20%0A%20%20%20%20%20%20%20%0A%20%20title%2C%0A%20%20subtitle%2C%0A%20%20'slug'%3A%20slug.current%2C%0A%20%20date%2C%0A%20%20'author'%3A%20author-%3E%7Bname%2C%20'avatar'%3A%20avatar.asset-%3Eurl%7D%2C%0A%20%20coverImage%2C%0A%0A%20%20%20%20%20%20content%5B%5D%7B...%2C%20%22asset%22%3A%20asset-%3E%7D%0A%20%20%20%20%7D&%24slug=%22first-tought%22",
    method: 'GET',
    headers: {
      date: 'Fri, 21 May 2021 16:36:38 GMT',
      'content-type': 'application/json; charset=utf-8',
      'x-ratelimit-limit-second': '500',
      'x-ratelimit-remaining-second': '499',
      'ratelimit-remaining': '499',
      'ratelimit-limit': '500',
      'ratelimit-reset': '1',
      'content-length': '97',
      vary: 'Origin',
      xkey: 'project-ulym7tod, project-ulym7tod-production',
      via: '1.1 google',
      'alt-svc': 'clear',
      connection: 'close'
    },
    statusCode: 401,
    statusMessage: 'Unauthorized'
  },
  statusCode: 401,
  responseBody: '{\n' +
    '  "errorCode": "SIO-401-ANF",\n' +
    '  "message": "Session not found",\n' +
    '  "statusCode": 401,\n' +
    '  "error": "Unauthorized"\n' +
    '}'
}

Not sure what is going on here but something is not right.
Any help will be much appreciated.

Thanks!

**Comment by flayks** [comment]
I've got the same error when updating to `@sanity/desk-tool` to 2.10.4

**Comment by aaronadamsCA** [comment]
We are seeing the same issue. Our environment:

```sh
$ yarn santy versions
@sanity/cli                2.10.0 (up to date)
@sanity/base               2.10.2 (up to date)
@sanity/cli                2.10.0 (up to date)
@sanity/core               2.10.2 (up to date)
@sanity/dashboard           2.7.0 (up to date)
@sanity/default-layout     2.10.2 (up to date)
@sanity/default-login       2.8.0 (up to date)
@sanity/desk-tool          2.10.4 (up to date)
@sanity/google-maps-input  2.10.2 (up to date)
@sanity/vision             2.10.0 (up to date)

$ npm -v && node -v
6.14.12
v14.16.1
```

**Comment by aaronadamsCA** [comment]
Here is a screenshot of the error location:

![image](https://user-images.githubusercontent.com/1505561/119358818-4949cb80-bca9-11eb-8c07-f679264c13c3.png)

It looks like CommonJS files aren't seeing a named export `hues` from the package `@sanity/color`.

**Comment by levi-l-damian** [comment]
Here are my settings.
```
$ yarn sanity versions
yarn run v1.22.10

@sanity/cli                  2.2.6 (latest: 2.10.5)
@sanity/base                2.10.5 (up to date)
@sanity/cli                  2.2.6 (latest: 2.10.5)
@sanity/code-input           2.9.0 (up to date)
@sanity/components           2.2.6 (up to date)
@sanity/core                2.10.5 (up to date)
@sanity/default-layout      2.10.5 (up to date)
@sanity/default-login       2.10.5 (up to date)
@sanity/desk-tool           2.10.5 (up to date)
@sanity/production-preview   2.2.6 (up to date)
@sanity/vision              2.10.5 (up to date)

$ npm -v && node -v
6.14.13
v14.17.0
```

**Comment by chrisvince** [comment]
I am having this same issue.

**Comment by rexxars** [comment]
It looks like it might be using an old version of `@sanity/color`, for some reason... Could you try running `yarn why @sanity/color` and pasting the output?

**Comment by browniebroke** [comment]
I think I have the same problem, here is mine:

```
 ➜ yarn why @sanity/color
yarn why v1.22.10
[1/4] 🤔  Why do we have the module "@sanity/color"...?
[2/4] 🚚  Initialising dependency graph...
[3/4] 🔍  Finding dependency...
[4/4] 🚡  Calculating file sizes...
=> Found "@sanity/color@2.0.15"
info Has been hoisted to "@sanity/color"
info Reasons this module exists
   - Hoisted from "@sanity#types#@sanity#color"
   - Hoisted from "@sanity#base#@sanity#color"
   - Hoisted from "@sanity#desk-tool#@sanity#form-builder#@sanity#color"
info Disk size without dependencies: "200KB"
info Disk size with unique dependencies: "200KB"
info Disk size with transitive dependencies: "200KB"
info Number of shared dependencies: 0
=> Found "sanity-plugin-media#@sanity/color@2.0.12"
info This module exists because "sanity-plugin-media" depends on it.
info Disk size without dependencies: "160KB"
info Disk size with unique dependencies: "160KB"
info Disk size with transitive dependencies: "160KB"
info Number of shared dependencies: 0
=> Found "@sanity/ui#@sanity/color@2.0.17"
info This module exists because "@sanity#ui" depends on it.
info Disk size without dependencies: "368KB"
info Disk size with unique dependencies: "368KB"
info Disk size with transitive dependencies: "368KB"
info Number of shared dependencies: 0
=> Found "sanity-plugin-media#@sanity/ui#@sanity/color@2.0.15"
info This module exists because "sanity-plugin-media#@sanity#ui" depends on it.
info Disk size without dependencies: "200KB"
info Disk size with unique dependencies: "200KB"
info Disk size with transitive dependencies: "200KB"
info Number of shared dependencies: 0
✨  Done in 0.91s.
```

**Comment by rexxars** [comment]
Could you try reinstalling dependencies by doing  `rm -rf node_modules yarn.lock && yarn` (or `rm -rf node_modules yarn.lock && npm i` if you prefer npm)?

**Comment by levi-l-damian** [comment]
This one fixed the problem, many thanks.
`$ rm -rf node_modules yarn.lock && npm i`

Many thnaks!

**Comment by browniebroke** [comment]
Fixed it for me as well, thanks!

**Comment by aaronadamsCA** [comment]
I'm gonna go out on a limb here and assume everyone affected is on Yarn v1. If so, this is almost certainly a Yarn v1 bug.

The only way I could resolve this locally was to delete `node_modules/`. The only way I could resolve it on Netlify was to manually clear cache and deploy. That's not great! But I don't think Sanity is at fault here, because the dependencies all seem to be correctly specified.

Yarn reports `@sanity/color` v2 is the only one installed:

```
$ yarn why @sanity/color
yarn why v1.22.5
[1/4] Why do we have the module "@sanity/color"...?
[2/4] Initialising dependency graph...
[3/4] Finding dependency...
[4/4] Calculating file sizes...
=> Found "@sanity/color@2.1.0"
info Reasons this module exists
   - "_project_#@shopnishe#sanity-studio#@sanity#base" depends on it
   - Hoisted from "_project_#@shopnishe#sanity-studio#@sanity#base#@sanity#color"
   - Hoisted from "_project_#@shopnishe#sanity-studio#@sanity#desk-tool#@sanity#form-builder#@sanity#color"
   - Hoisted from "_project_#@shopnishe#sanity-studio#@sanity#base#@sanity#types#@sanity#color"
   - Hoisted from "_project_#@shopnishe#sanity-studio#@sanity#base#@sanity#ui#@sanity#color"
info Disk size without dependencies: "252KB"
info Disk size with unique dependencies: "252KB"
info Disk size with transitive dependencies: "252KB"
info Number of shared dependencies: 0
Done in 0.88s.
```

Yet when I look in `node_modules/` I can clearly see @sanity/color v1 hiding out at `node_modules/@sanity/base/node_modules/@sanity/color`. This causes Node.js module resolution to use v1, even though Yarn claims only v2 is currently installed.

This is why clearing and reinstalling all modules is the only working solution. Sigh.

TLDR, Yarn v1 sucks and is no longer supported; upgrade to Yarn v2 or switch to NPM v7.

**Comment by RobEasthope** [comment]
Just tested @rexxars fix but with no luck. Ditching Yarn and using NPM solved the issue for me
